packages/block-editor/src/components/block-list/block-wrapper.js

```diff
@@ -265,6 +275,60 @@ export function useBlockWrapperProps( props = {}, { __unstableIsHtml } = {} ) {
 		};
 	}, [ isNavigationMode, isHovered, setHovered ] );
 
+	useEffect( () => {
+		if ( isFullSiteEditingActive ) {
+			// To accurately determine which blocks are hovered we must look at the elements under the cursor.
+			// Internal block inserters will fire events like mouseleave for blocks they are visually contained within,
+			// which makes an add/remove by clientId per individual block approach inaccurate.
+			function getHoveredBlocksFromCursor( event ) {
```

Comment:
Can move this outside of the `useBlockWrapperProps`? I don't see any references that could stop us from doing it. Though it may feel weird if it's moved from here because it makes it less easier to read. We need to lookup the function.

Reply:
Whats the thought with that move?  That it could be exported and used elsewhere?  Im open to the idea of moving, although Im not exactly sure where to move it and not very convinced it should.  But Id also think if we were to move it we should tie in everything with the timestamps/whatever else we need to limit the unnecessary recalculations? 🤔

Reply:
It's my personal preference, but usually I try to avoid declaring functions inside functions if the inner function isn't using any variables/references from the outer scope. We can make the `useBlockWrapperProps` a bit shorter and less noisy.

Also, if we move `getHoveredBlocksFromCursor` function outside, then we will have only one reference to this function, rather than one reference to each `useBlockWrapperProps` call.

TLDR: Micro optimization and personal preference :D 

`evaluateHoveredBlocks` is tied to the outer scope so we can't move that though.

Reply:
Yeah i definitely agree.  I moved both functions to another file to clean it up a bit.